subsys/bluetooth/mesh/scheduler_internal.h

```diff
@@ -18,6 +18,19 @@
 extern "C" {
 #endif
 
+/* Returns true if the action can be packed without its data being compromised, false otherwise */
+static inline bool schedule_action_sanitize(const struct bt_mesh_schedule_entry *entry, uint8_t idx)
+{
+	if (entry->year > 127 || entry->month > 4095 || entry->day > 31 || entry->hour > 31 ||
+	    entry->minute > 63 || entry->second > 63 || entry->day_of_week > 127 ||
+	    entry->action > 15 || entry->transition_time > 255 ||
+		entry->scene_number > 65535 || idx > 15) {
```

Comment:
If including this in `scheduler_action_pack`, I will make the function return its success state then? Non-void.

Reply:
Yes, return bool. The current contract for the pack function is "pack these values if valid, otherwise pack some undefined values". A better contract would be "pack these values and return true if valid, return false and do not touch the buffer otherwise"

Reply:
> [...] Also, please create #defines for all these values (preferably defined as a computation on number of bits for the fields which use a full N-bit range)

^ This is not handled.